# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/sandbox/linux/launch/SandboxLaunch.cpp
# Commit: 493b443954fe
# Full Hash: 493b443954fe15f7b542ba14671f25e5f8531dff
# Author: Jed Davis <jld@mozilla.com>
# Date: 2019-02-28 10:55:23
# Regressor Bug: 1506291
# File Overlap Count: 1
# Description:
#   Bug 1506291 - Add Linux sandboxing for the RDD (media decoder) process. r=gcp,mjf,flod
#   
#   The seccomp-bpf policy is currently just the "common" policy with no
#   additions (but with the fixes in bug 1511560 to enable shared memory
#   creation).  The file broker policy allows shared memory creation and
# ==============================================================================

diff -r bf58d8320f5a -r 493b443954fe security/sandbox/linux/launch/SandboxLaunch.cpp
--- a/security/sandbox/linux/launch/SandboxLaunch.cpp	Mon Feb 25 16:20:50 2019 +0000
+++ b/security/sandbox/linux/launch/SandboxLaunch.cpp	Wed Feb 27 20:14:54 2019 +0000
@@ -228,6 +228,8 @@
       }
       return 0;
 #endif
+    case GeckoProcessType_RDD:
+      return PR_GetEnv("MOZ_DISABLE_RDD_SANDBOX") == nullptr ? 1 : 0;
     default:
       return 0;
   }
@@ -277,12 +279,13 @@
   switch (aType) {
 #ifdef MOZ_GMP_SANDBOX
     case GeckoProcessType_GMPlugin:
+#endif
+    case GeckoProcessType_RDD:
       if (level >= 1) {
         canChroot = true;
         flags |= CLONE_NEWNET | CLONE_NEWIPC;
       }
       break;
-#endif
 #ifdef MOZ_CONTENT_SANDBOX
     case GeckoProcessType_Content:
       if (level >= 4) {